### PR TITLE
Show level as practice level depending on the type

### PR DESCRIPTION
**Pull request by gabriellui1**
fixes ENG-911
Show some level as practice levels and some levels as challenge levels.
image

## Summary by CodeRabbit

- **Enhancements**
  - Updated the progress icon logic for improved consistency.
  - Enhanced flexibility in the ModuleRow component by updating property requirements and default values.
  - Improved data retrieval for campaign levels to include assessment information.

### CODE REVIEW

File: app/views/parents/student-progress/CocoModuleProgressComponent.vue [hunk 1]
```diff
@@ -77,7 +77,7 @@ export default {
   },
   methods: {
     getIconType (level) {
-      if (level.kind === 'practice' || level.practice) return 'practicelvl'
+      if (level.kind === 'practice' || level.practice !== false) return 'practicelvl'
```
Comment (smallst) [inline app/views/parents/student-progress/CocoModuleProgressComponent.vue:80]: https://github.com/codecombat/codecombat/blob/b16891cb3b60b6c4d17d2454a419e390c2ed07eb/app/core/utils.js#L1603-L1615 you can follow this logic to check practice or challenge. note that all `assessment` lead to challenge.

**Review by smallst — CHANGES_REQUESTED** [review]

File: app/views/parents/student-progress/CocoModuleProgressComponent.vue (hunk 1 above)
Comment (smallst) [reply app/views/parents/student-progress/CocoModuleProgressComponent.vue:80]: and if a level not match in practice or challenge we can also show as `challenge` icon

Comment (gabriellui1) [reply app/views/parents/student-progress/CocoModuleProgressComponent.vue]: updated! level.practice always seems to return false or undefined so level.kind is used instead. I'm curious, what even broke the code in the first place?

**Review by mrfinch — COMMENTED** [review]
there are vue warnings for levelNumber and displayName. Can we pass empty strings for now?
Screenshot 2024-07-29 at 12 02 25 PM

**Review by mrfinch — APPROVED** [review]

**Pull request merged**